.github/workflows/odh-kfto-sdk-notebooks-sync.yaml

```diff
@@ -0,0 +1,162 @@
+            if echo "$versions" | grep -q "${TRAINING_SDK_RELEASE_VERSION}"; then
+                echo "Version ${TRAINING_SDK_RELEASE_VERSION} is available for $package_name"
+                directories+=($(grep --exclude-dir=.git --exclude-dir=.github --exclude-dir=intel --exclude-dir=tensorflow --exclude-dir=rocm-tensorflow --include="Pipfile*" -rl "$package_name = \"==[0-9.]*\"" | xargs dirname | sort | uniq))
+                counter=0
+                total=${#directories[@]}
+                echo -----------
+                for dir in "${directories[@]}"; do
+                  counter=$((counter+1))
+                  echo "--Processing directory $counter '$dir' of total $total"
+                  cd "$dir"
+                  minimum_supported_python_version_major=$(echo "${MINIMUM_SUPPORTED_PYTHON_VERSION}" | awk -F '.' '{print $1}') #integer of MINIMUM_SUPPORTED_PYTHON_VERSION env variable
+                  minimum_supported_python_version_minor=$(echo "${MINIMUM_SUPPORTED_PYTHON_VERSION}" | awk -F '.' '{print $2}') #decimal of MINIMUM_SUPPORTED_PYTHON_VERSION env variable
+                  if ! [ -f "Pipfile" ]; then
+                      if [ -f "Pipfile.cpu" ]; then
+                          pipfile_python_version=$(grep -E '^python_version' ./Pipfile.cpu | cut -d '"' -f 2) # extracted from pipfile.cpu
+                      fi
+                  else
+                      pipfile_python_version=$(grep -E '^python_version' ./Pipfile | cut -d '"' -f 2) # extracted from pipfile
+                  fi
+                  pipfile_python_version_major=$(echo "$pipfile_python_version" | awk -F '.' '{print $1}')
+                  pipfile_python_version_minor=$(echo "$pipfile_python_version" | awk -F '.' '{print $2}')
+                  if [[ " ${available_python_versions[@]} " =~ " ${pipfile_python_version} " && "$pipfile_python_version_major" -ge "$minimum_supported_python_version_major" && "$pipfile_python_version_minor" -ge "$minimum_supported_python_version_minor" ]]; then
+                      if ! [ -f "Pipfile" ]; then
+                          if [ -f "Pipfile.cpu" ]; then
+                              install_package_using_pipenv Pipfile.cpu Pipfile.lock.cpu
+                          fi
+                          if [ -f "Pipfile.gpu" ]; then
+                              install_package_using_pipenv Pipfile.gpu Pipfile.lock.gpu
+                          fi
+                      else
+                          #install specified package
+                          install_package_using_pipenv
+                      fi
+                      else
+                      echo "Skipped installation of ${package_name} with version ${TRAINING_SDK_RELEASE_VERSION} in $dir"
+                  fi
+                  cd -
+                  echo "$((total-counter)) directories remaining.."
+                done
+                # Refresh Pipfile.Lock files
+                cd $REPO_NAME && make refresh-pipfilelock-files && cd -
```

Comment:
yes, using our make target should be more robust and produce same results as our https://github.com/opendatahub-io/notebooks/blob/23041ca5bb472587736074f9963dc52199377d7b/.github/workflows/piplock-renewal.yaml#L69 does

/lgtm